Tests/LegacyUnit/FrontEnd/DefaultControllerTest.php

```diff
@@ -6604,6 +6604,7 @@ public function registrationFormHtmlspecialcharsEventTitle()
         );
 
         $this->subject->piVars['seminar'] = $eventUid;
+        $this->subject->piVars['action'] = 'register';
```

Comment:
Why the changes in this file?

Reply:
Because otherwise [$registrationEditor->setAction($this->piVars['action']);](https://github.com/oliverklee/ext-seminars/blob/main/Classes/FrontEnd/DefaultController.php#L3016) will be empty and `setAction()` says in the header comment 
> * @param string $action action for which to create the form, must be either "register" or "unregister", must not be empty

Reply:
Then we need to fix the call to `setAction` instead, do a `??` check and a string cast around that.

Reply:
Like so 

```
        $action = $this->piVars['action'] ?? 'register';
        $registrationEditor->setAction((string) $action);
```

Reply:
Yes, or like this:
```php
$action = (string)($this->piVars['action'] ?? 'register');
$registrationEditor->setAction($action);
```